dds_web/user.py

```diff
@@ -22,29 +23,48 @@ def login():
     if request.method == "GET":
         if session.get("is_admin"):
             return redirect(url_for("admin.admin_page"))
-        elif session.get("current_user") and session.get("usid"):
+        elif session.get("current_user"):
             return redirect(url_for("user.user_page", loginname=session["current_user"]))
         else:
             return render_template("user/login.html", next=request.args.get("next"))
 
+    # Login form procedure
     if request.method == "POST":
+        # Get username and password from form
         username = request.form.get("username")
         password = request.form.get("password")
-        credentials_validated, is_facility, message, user_info = validate_user_credentials(
-            username, password
-        )
-        if not credentials_validated:
+
+        # Authenticate user
+        try:
+            user_ok = dds_auth.verify_user_pass(username=username, password=password)
+        except sqlalchemy.exc.SQLAlchemyError as sqlerr:
+            # TODO (ina): Create custom error page
+            app.logger.exception(sqlerr)
+            return str(sqlerr), 500
+        except exceptions.AuthenticationError as autherr:
+            app.logger.exception(autherr)
             return render_template(
-                "user/login.html", next=request.form.get("next"), login_error_message=message
+                "user/login.html", next=request.form.get("next"), login_error_message=str(autherr)
             )
```

Comment:
From line 44 to 48 is the next except block from the previous one in the previous comment. For login related errors, one usually would like to show a highlighted or popped-up error text/bubble instead of returning a custom error page.

Reply:
Ah, right, yes. Will fix!